src/main/scala/is/hail/rvd/OrderedRVD.scala

```diff
@@ -540,27 +542,28 @@ object OrderedRVD {
     val min = pkis.map(_.min).min(pkOrd)
     val max = pkis.map(_.max).max(pkOrd)
 
-    keys = min +: keys :+ max
-
     val ab = new ArrayBuilder[RegionValue]()
-    var i = 0
-    while (i < keys.length) {
-      if (i == 0
-        || pkOrd.compare(keys(i - 1), keys(i)) != 0)
-        ab += keys(i)
+    ab += min
+    var start = 0
+    while (start < keys.length
+      && pkOrd.compare(min, keys(start)) == 0)
+      start += 1
+    var i = 1
+    while (i < nPartitions && start < keys.length) {
+      var end = ((i.toDouble * keys.length) / nPartitions).toInt
+      if (start > end)
+        end = start
+      while (end < keys.length - 1
+        && pkOrd.compare(keys(end), keys(end + 1)) == 0)
+        end += 1
+      ab += keys(end)
+      start = end + 1
       i += 1
     }
-    keys = ab.result()
-
-    // FIXME weighted
-    val partitionMaxes =
-      if (keys.length <= nPartitions + 1)
-        keys
-      else {
-        val k = keys.length / nPartitions
-        assert(k > 0)
-        Array.tabulate(nPartitions)(i => keys(i * k)) :+ max
-      }
+    if (pkOrd.compare(ab.last, max) != 0)
+      ab += max
+    val partitionMaxes = ab.result()
```

Comment:
This contains the minimum too, more consistent to call it `rangeBounds` as used for the parameter in `OrderedRVDPartitioner.makeRangeBoundIntervals`?

Reply:
Good catch, I had meant to rename that `partitionEdges` (following the `hist`).  Done.